src/view/observer/clickobserver.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import DomEventObserver from './domeventobserver.js';
+
+/**
+ * {@link engine.view.Document#click Click} event observer.
+ *
+ * Note that this observer is not attached by the {@link engine.view.Document} and is not available by default.
```

Comment:
The only thing missing here is rich documentation, similar to [which comes with the `KeyObserver`](https://github.com/ckeditor/ckeditor5-engine/blob/t/577/src/view/observer/keyobserver.js#L39-L65).